autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/importing/DomainNameNormalizer.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.autofill.impl.importing
+
+import com.duckduckgo.autofill.api.domain.app.LoginCredentials
+import com.duckduckgo.autofill.impl.urlmatcher.AutofillUrlMatcher
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import javax.inject.Inject
+
+interface DomainNameNormalizer {
+    suspend fun normalizeDomains(unnormalized: List<LoginCredentials>): List<LoginCredentials>
+}
+
+@ContributesBinding(AppScope::class)
+class DefaultDomainNameNormalizer @Inject constructor(
+    private val urlMatcher: AutofillUrlMatcher,
+) : DomainNameNormalizer {
+    override suspend fun normalizeDomains(unnormalized: List<LoginCredentials>): List<LoginCredentials> {
+        return unnormalized.map {
+            val currentDomain = it.domain ?: return@map null
```

Comment:
>urlMatcher.cleanRawUrl(currentDomain) is returning a domain
- **yes**

>it.domain can contain a url
- **yes**
- normally, `LoginCredentials` would be a domain, but since we've brought in the external format it's temporarily a URL, and converting to a domain is what's happening to get the external format into the correct internal format

Reply:
That's great to know. So we are bringing an external set of credentials, which follow a similar structure but differ on the expected format of the values. And we are using our domain object to map them. That can be problematic. I would like to avoid reusing our domain object to map an external object we don't control values. There's risk of introducing bugs difficult to spot and wrong expectations when working with those objects.

I propose we define a data class where we store the external objects first. We can have GoogleCSVLoginCrendential, with url instead of domain as attribute. Then, they will be mapped into our domain object as part of the processing.